### PR TITLE
[ Keychron K8 ] Migrate folder structure

### PR DESCRIPTION
**Pull request by tbowmo**
## Description
Adding support for mac layout (using the switch on the side of the keyboard), grabbed the layout from C1
Tested that it did swap gui / alt keys as intended, when flipping the switch

Both ansi and iso layouts stick as close to the original as possible.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k8/rgb/config.h [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #define MATRIX_ROW_PINS { C15, D11, D10, D9, D8, D7 }
 
 // Connects each switch in the dip switch to the GPIO pin of the MCU
-#define DIP_SWITCH_PINS { D5, D6 }
+#define DIP_SWITCH_PINS { D5, D4 }
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/config.h]: Our convention so far is index 0 - OS, index 1 - BT ( to follow up on C series not having BT as an option)
the array should be flipped for consistency, and the keymaps as well

File: keyboards/keychron/k8/config.h [hunk 2]
```diff
@@ -18,4 +16,8 @@
 
 #pragma once
 
-#include "config_common.h"
+// Enable RGB Matrix
+#define VIA_QMK_RGBLIGHT_ENABLE
```
Comment (dexter93) [inline keyboards/keychron/k8/config.h]: that should probably live inside via keymap. see https://github.com/SonixQMK/qmk_firmware/pull/224/files#diff-d63594f4756161abc37655e1c965531f8ff18b90e1424e1ee601cb1c1ff0aa0cR22

File: keyboards/keychron/k8/k8.c [hunk 3]
```diff
@@ -17,3 +17,27 @@
  */
 
 #include "k8.h"
+
+bool dip_switch_update_kb(uint8_t index, bool active){
```
Comment (dexter93) [inline keyboards/keychron/k8/k8.c]: it is much neater to have this inside a keymap. allows for easier user customizability, at least for the OS switch

We might have to revisit this when BT drops and seperate them. Talks around it are probably needed.

File: keyboards/keychron/k8/k8.h [hunk 4]
```diff
@@ -25,3 +25,19 @@
 #elif defined(KEYBOARD_keychron_k8_white)
     #include "white.h"
 #endif
+
+enum layer_names {
```
Comment (dexter93) [inline keyboards/keychron/k8/k8.h]: defining layer names and custom keycodes here is a nono. They should all live in a keymap,  even if duplicate

File: keyboards/keychron/k8/keymaps/via/keymap.c [hunk 5]
```diff
@@ -44,28 +45,28 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
         KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,                                      KC_RALT, KC_RGUI, MO(1),   KC_RCTL,     KC_LEFT, KC_DOWN, KC_RGHT
     ),
     [1] = LAYOUT_tkl_ansi(
```
Comment (dexter93) [inline keyboards/keychron/k8/keymaps/via/keymap.c:47]: this implies that via will have 1 layer active and 3 blank. If we're going to use ansi by default, add the whole thing. Saves hassle on debugging and all our keycodes are via compatible anyway

File: keyboards/keychron/k8/rgb/ansi/config.h [hunk 6]
```diff
@@ -18,3 +18,6 @@
 
 // RGB Matrix Config
 #define DRIVER_LED_TOTAL 87
+
+/* USB Device descriptor parameter */
+#define PRODUCT_ID      0xFE0E
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/ansi/config.h:23]: it is safe to do the device version here too. see https://github.com/SonixQMK/qmk_firmware/pull/224/files#diff-de75e964a9aba6ca71121c5c23174ec91a885eefea2f554dc054c54dd1598af9R19

File: keyboards/keychron/k8/rgb/config.h [hunk 7]
```diff
@@ -18,15 +18,15 @@
 
 #pragma once
 
-#include "config_common.h"
-
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0x3434
-#define PRODUCT_ID      0xFE0E
+// PRODUCT_ID is specified separately for ISO / ANSI models
 #define DEVICE_VER		0x0001
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/config.h]: this needs to be moved for consistency. also the tabs should be space here

File: keyboards/keychron/k8/rgb/optical_ansi/config.h [hunk 8]
```diff
@@ -17,3 +17,10 @@
 
 // RGB Matrix Config
 #define DRIVER_LED_TOTAL 87
+
+/* USB Device descriptor parameter */
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/optical_ansi/config.h:21]: pid, ver and driver total are only needed here. the rest can be in rgb/config.h as common

File: keyboards/keychron/k8/config.h [hunk 9]
```diff
@@ -18,4 +16,8 @@
...
+
+#define USB_POLLING_INTERVAL_MS 1
+#define QMK_KEYS_PER_SCAN 4
```
Comment (dexter93) [inline keyboards/keychron/k8/config.h]: should these be shipped by default?
should they go in a keymap/user directory?

Don't really know

File: keyboards/keychron/k8/keymaps/iso/keymap.c [hunk 10]
```diff
@@ -19,78 +20,73 @@
...
 				| ESC |  | F1 | F2 | F3 | F4 | | F5 | F6 | F7 | F8 | | F9| F10| F11| F12|  | |PSCR|????|PAUS|
 				+--------------------------------------------------------------------------+------|----|----|
  				|  ~  |  1 |  2 |  3 |  4 |  5 |  6 |  7 |  8 |  9 |  0 |  - |  = | BACKSP | |INS |SCRL|PGUP|
  				+--------------------------------------------------------------------------+------|----|----|
- 				|  TAB  |  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P |  [ |  ] |   \  | |DEL |END |PGDN|
- 				+--------------------------------------------------------------------------+------|----|----|
- 				| CAPSLCK  |  A |  S |  D |  F |  G |  H |  J |  K |  L | ; | ' |  RETURN  |                |
+        |  TAB  |  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P |  [ |  ] |      | |DEL |END |PGDN|
+        +--------------------------------------------------------------------  RET | +--------------+
+        | CAPSLCK  |  A |  S |  D |  F |  G |  H |  J |  K |  L | ; | ' |  # |     |                |
```
Comment (dexter93) [inline keyboards/keychron/k8/keymaps/iso/keymap.c]: identations should be space, or this magnificent thing happens

**Review by dexter93 — COMMENTED** [review]
Shaping up nicely

File: keyboards/keychron/k8/keymaps/iso/keymap.c (hunk 10 above)
Comment (tbowmo) [reply keyboards/keychron/k8/keymaps/iso/keymap.c]: Darn.. Its vsCode that do some weird auto detection on space / tabs..

File: keyboards/keychron/k8/k8.c (hunk 3 above)
Comment (tbowmo) [reply keyboards/keychron/k8/k8.c]: I know, but I just felt it was a bit silly to have multiple definitions of the same code, specially, when we do call the _user variant.

If you look at the qmk docs, they do say, that we can put in keyboard level functions here..

But won't let it be a breaking thing. Will move it back to the keymaps.

File: keyboards/keychron/k8/config.h (hunk 9 above)
Comment (tbowmo) [reply keyboards/keychron/k8/config.h]: I do believe they where coming in from the xelus pr, and perhaps moved a bit around in my refactoring / rebase / merge efforts

File: keyboards/keychron/k8/keymaps/iso/keymap.c (hunk 10 above)
Comment (tbowmo) [reply keyboards/keychron/k8/keymaps/iso/keymap.c]: Should be fixed in keymaps now..

File: keyboards/keychron/k8/rgb/optical_ansi/config.h (hunk 8 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/optical_ansi/config.h:21]: Removed "obsolete" defines

File: keyboards/keychron/k8/rgb/config.h (hunk 1 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/config.h]: I changed it, both here and all keymaps..

File: keyboards/keychron/k8/config.h (hunk 9 above)
Comment (tbowmo) [reply keyboards/keychron/k8/config.h]: I removed the two..

**Pull request merged**